src/main/java/org/craftercms/studio/impl/v2/repository/blob/s3/StudioAwsS3BlobStore.java

```diff
@@ -361,4 +358,41 @@ public void initialPublish(String siteId) throws SiteNotFoundException {
         }
     }
 
+    @Override
+    public void publishAll(String siteId, String publishingTarget) {
+        // this method should not be called
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public RepositoryChanges preparePublishAll(String siteId, String publishingTarget) {
+        // this method should not be called
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void completePublishAll(String siteId, String publishingTarget, RepositoryChanges changes) {
```

Comment:
Wasn't there a way in S3 to bulk copy and delete?

Reply:
there is a bulk delete, but no bulk copy

Reply:
do you think we would gain a performance improvement if we use the bulk delete?

Reply:
it might change a lot, probably there will be improvement only if a publish contains a lot of deleted files

Reply:
@sumerjabri what do you think?